internal/tiltfile/extension/store.go

```diff
@@ -0,0 +1,71 @@
+package extension
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/windmilleng/tilt/internal/tiltfile"
+)
+
+const extensionDirName = "tilt_modules"
+
+type Store interface {
+	// Stat is used to check if an extension exists before fetching it
+	Stat(ctx context.Context, moduleName string) (string, error)
+	Write(ctx context.Context, contents ModuleContents) (string, error)
+}
+
+// TODO(dmiller): should this include an integrity hash?
+type ModuleContents struct {
+	Name             string
+	TiltfileContents string
+	GitCommitHash    string
+	Source           string
+	TimeFetched      time.Time
+}
+
+type LocalStore struct {
+	baseDir string
+}
+
+func NewLocalStore(baseDir string) *LocalStore {
+	return &LocalStore{
+		baseDir: filepath.Join(baseDir, extensionDirName),
+	}
+}
+
+func (s *LocalStore) Stat(ctx context.Context, moduleName string) (string, error) {
+	tiltfilePath := filepath.Join(s.baseDir, moduleName, tiltfile.FileName)
+
+	_, err := os.Stat(tiltfilePath)
+	if err == nil {
+		return tiltfilePath, nil
+	}
+
+	if os.IsNotExist(err) {
+		return "", nil
+	}
+
+	return "", err
+}
+
+func (s *LocalStore) Write(ctx context.Context, contents ModuleContents) (string, error) {
+	moduleDir := filepath.Join(s.baseDir, contents.Name)
+	if err := os.MkdirAll(moduleDir, os.FileMode(0700)); err != nil {
+		return "", fmt.Errorf("couldn't store module %s: %v", contents.Name, err)
```

Comment:
mind testing manually and/or in a unit test that the error message for this and below include the path that we failed to write?

it'd be super annoying if it were just like "couldn't store module openfaas: permission denied" and there was no pointer to what permissions needed to be fixed

also....errors.Wrapf!

Reply:
I'll just manually include it.